test/controller/legacy_pages_controller_test.rb

```diff
@@ -37,18 +37,15 @@ class LegacyPagesControllerTest < ActionController::TestCase
     assert_nil session[:_turbolinks_redirect_to]
   end
 
-  test "set_xhr_redirected_to sets X-XHR-Redirected-To" do
-    get :index, {}, {_turbolinks_redirect_to: 'http://test.host/expected'}
-    assert_response :ok
-    assert_equal 'http://test.host/expected', response.headers['X-XHR-Redirected-To']
+  test "XHR get for the page matching the redirect will set X-XHR-Redirected-To" do
+    session[:_turbolinks_redirect_to] = 'http://test.host/legacy_pages/321'
+    get :show, params: {id: 321}, xhr: true
```

Comment:
You can set the session in this line as well if you think it would be more clear:

http://api.rubyonrails.org/v4.2.7.1/classes/ActionController/TestCase/Behavior.html#method-i-get
![image](https://cloud.githubusercontent.com/assets/325/18961847/6c3ede8a-863c-11e6-908e-91f9ade0d6a3.png)